### PR TITLE
image prompts - Entrypoint prompt - additional CLI argument

### PR DESCRIPTION
**Pull request by ATheorell**
This is a continuation of PR https://github.com/gpt-engineer-org/gpt-engineer/pull/1066, which in addition to providing command line arguments to provide a custom prompt file and a directory with images, also allows the user to pass a prompt to generate entry point.
I suggest that we continue with https://github.com/gpt-engineer-org/gpt-engineer/pull/1066 here @TheoMcCabe

Unfortunately, the PR has drifted a bit and now contains substantial changes

### CODE REVIEW

**Comment by ATheorell** [comment]
If someone has an idea why this particular test fails, I would love to hear it. The case:

The test passes when running only ´´´ test_ai.py``` through pytest, but when running all the test with pytest at once it fails.

File: gpt_engineer/applications/cli/cli_agent.py [hunk 1]
```diff
@@ -205,19 +208,18 @@ def improve(
...
-
-        # No need to run entrypoint for improve right?
-        # if not execution_command and ENTRYPOINT_FILE not in files_dict:
-        #     entrypoint = gen_entrypoint(
-        #         self.ai, files_dict, self.memory, self.preprompts_holder
-        #     )
-        #     combined_dict = {**files_dict, **entrypoint}
-        #     files_dict = FilesDict(combined_dict)
-
+        # entrypoint = gen_entrypoint(
```
Comment (TheoMcCabe) [inline gpt_engineer/applications/cli/cli_agent.py:211]: are we sure we want commented code here? its always in the git history if we want to bring it back later - or are we expecting this to be commented in and out regularly

File: gpt_engineer/applications/cli/main.py [hunk 2]
```diff
@@ -213,6 +285,14 @@ def main(
         The endpoint for Azure OpenAI services.
     use_custom_preprompts : bool
         Flag indicating whether to use custom preprompts.
+    prompt_file : str
+        Relative path to a text file containing a prompt.
+    entrypoint_prompt_file: str
+        Relative path to a text file containing a file that specifies requirements for you entrypoint.
```
Comment (TheoMcCabe) [inline gpt_engineer/applications/cli/main.py:291]: great

File: gpt_engineer/core/default/steps.py [hunk 3]
```diff
@@ -302,17 +318,17 @@ def _improve_loop(
...
-                HumanMessage(
-                    content="Some previously produced diffs were not on the requested format, or the code part was not found in the code. Details: "
-                    + "\n".join(problems)
-                    + "\n Please FOCUS ON the problematic diffs, making sure that the failing ones are now on the correct format and can be found in the code. Make sure to not repeat past mistakes. \n"
-                )
-            )
-            messages = ai.next(messages, step_name=curr_fn())
-            edit_refinements += 1
-            files_dict = salvage_correct_hunks(messages, files_dict, problems)
+        # if len(problems) > 0:
```
Comment (TheoMcCabe) [inline gpt_engineer/core/default/steps.py:321]: same point about code comments

File: tests/caching_ai.py [hunk 4]
```diff
@@ -65,6 +69,7 @@ def next(
             cache = dict()
 
         messages_key = self.serialize_messages(messages)
+        print(messages_key)
```
Comment (TheoMcCabe) [inline tests/caching_ai.py:72]: probably an accident by me

**Review by TheoMcCabe — APPROVED** [review]

**Comment by TheoMcCabe** [comment]
OK @ATheorell  Ive added some docs - and ive tested that its all working. looks good to me now. Im happy to merge - will leave the final review and merge to you?

**Pull request merged**